### ISSUE TITLE
AppCenter not detecting updates

### ISSUE THREAD

**Issue by jwinnie** [elementary/appcenter#439]
When I open AppCenter, it shows that all apps are up to date and the spinner stops spinning. However, `apt` says that there are updates. Relevant logs/screenshots are below. 

**Screenshot**
![screenshot from 2017-08-04 14 53 22](https://user-images.githubusercontent.com/20506149/28988492-e410ed32-7924-11e7-8684-c7e2069aea99.png)

<details>
<summary><b>Output of <code>appcenter --debug</code></b></summary>
<br>
<pre><code>
[INFO 14:53:45.316504] Application.vala:153: App Center version: 0.2.4
[INFO 14:53:45.316565] Application.vala:155: Kernel version: 4.4.0-83-generic
[DEBUG 14:53:45.321052] [PackageKit] adding state 0x1f418c0
[DEBUG 14:53:45.321180] [PackageKit] role now get-packages
[DEBUG 14:53:45.381251] [PackageKit] emit transaction-list-changed
[DEBUG 14:53:46.390100] [PackageKit] emit transaction-list-changed
[DEBUG 14:53:46.390183] [PackageKit] remove state 0x1f418c0
[WARNING 14:53:46.412579] [Gtk] Theme parsing error: gtk-widgets.css:2045:14: 'min-height' is not a valid property name
[WARNING 14:53:46.413123] [Gtk] Theme parsing error: gtk-widgets.css:2046:13: 'min-width' is not a valid property name
[WARNING 14:53:46.413233] [Gtk] Theme parsing error: gtk-widgets.css:2051:13: 'min-width' is not a valid property name
[DEBUG 14:53:46.437607] [Gtk] Connecting to session manager
[DEBUG 14:53:46.443704] Client.vala:513: update cache called true
[DEBUG 14:53:46.443747] Client.vala:539: New refresh task
[DEBUG 14:53:46.443781] [PackageKit] adding state 0x1f41980
[DEBUG 14:53:46.443817] [PackageKit] role now refresh-cache
[DEBUG 14:53:46.444718] [PackageKit] notify::connected
[DEBUG 14:53:55.455952] [PackageKit] remove state 0x1f41980
[FATAL 14:53:55.456107] Client.vala:549: Update_cache: Refesh cache async failed - W: http://i-hate-farms.github.io/spores/dists/trusty/InRelease: Signature by key 2F3C3D45B6C9A4E9E22A1AE1F7F707E5B5329C31 uses weak digest algorithm (SHA1)W: The repository 'http://ppa.launchpad.net/mpstark/elementary-tweaks-daily/ubuntu xenial Release' does not have a Release file.W: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.W: See apt-secure(8) manpage for repository creation and user configuration details.
[DEBUG 14:55:30.823534] Houston.vala:51: Requesting newest applications from https://developer.elementary.io/api/newest/project
[DEBUG 14:55:30.823804] Houston.vala:51: Requesting newest applications from https://developer.elementary.io/api/newest/release
[DEBUG 14:55:30.823874] Houston.vala:51: Requesting newest applications from https://developer.elementary.io/api/newest/downloads
[DEBUG 14:55:30.831359] Settings.vala:163: Loading settings from schema 'io.elementary.appcenter.settings'
[DEBUG 14:55:30.832389] [PackageKit] adding state 0x38863d0
[DEBUG 14:55:30.832491] [PackageKit] role now get-packages
[DEBUG 14:55:31.767942] [PackageKit] adding state 0x7fcba8004800
[DEBUG 14:55:31.768006] [PackageKit] role now search-name
[DEBUG 14:55:31.769284] [PackageKit] adding state 0x7fcba8004e00
[DEBUG 14:55:31.769363] [PackageKit] role now search-name
[DEBUG 14:55:32.278764] [PackageKit] adding state 0x7fcbac005cb0
[DEBUG 14:55:32.278833] [PackageKit] role now search-name
[DEBUG 14:55:34.338827] [PackageKit] remove state 0x38863d0
[DEBUG 14:55:35.002748] [PackageKit] remove state 0x7fcba8004800
[DEBUG 14:55:35.002877] [PackageKit] adding state 0x7fcba80048c0
[DEBUG 14:55:35.002927] [PackageKit] role now get-details
[DEBUG 14:55:35.699859] [PackageKit] remove state 0x7fcba8004e00
[DEBUG 14:55:35.699959] [PackageKit] adding state 0x7fcba8004ec0
[DEBUG 14:55:35.700001] [PackageKit] role now get-details
[DEBUG 14:55:36.533035] [PackageKit] remove state 0x7fcbac005cb0
[DEBUG 14:55:36.533138] [PackageKit] adding state 0x7fcbac005d70
[DEBUG 14:55:36.533179] [PackageKit] role now get-details
[DEBUG 14:55:37.207781] [PackageKit] remove state 0x7fcba80048c0
[DEBUG 14:55:37.211017] [PackageKit] adding state 0x7fcba8004980
[DEBUG 14:55:37.211554] [PackageKit] role now search-name
[DEBUG 14:55:37.853055] [PackageKit] remove state 0x7fcba8004ec0
[DEBUG 14:55:37.854188] [PackageKit] adding state 0x7fcbac005830
[DEBUG 14:55:37.854232] [PackageKit] role now search-name
[DEBUG 14:55:38.449373] [PackageKit] remove state 0x7fcbac005d70
[DEBUG 14:55:38.449919] [PackageKit] adding state 0x7fcbac005e30
[DEBUG 14:55:38.449968] [PackageKit] role now search-name
[DEBUG 14:55:39.234928] [PackageKit] remove state 0x7fcba8004980
[DEBUG 14:55:39.235030] [PackageKit] adding state 0x7fcba8004a40
[DEBUG 14:55:39.235073] [PackageKit] role now get-details
[DEBUG 14:55:39.921541] [PackageKit] remove state 0x7fcbac005830
[DEBUG 14:55:39.921804] [PackageKit] adding state 0x7fcbac0058f0
[DEBUG 14:55:39.921914] [PackageKit] role now get-details
[DEBUG 14:55:40.597612] [PackageKit] remove state 0x7fcbac005e30
[DEBUG 14:55:40.597712] [PackageKit] adding state 0x7fcbac005ef0
[DEBUG 14:55:40.597753] [PackageKit] role now get-details
[DEBUG 14:55:41.239902] [PackageKit] remove state 0x7fcba8004a40
[DEBUG 14:55:41.242416] [PackageKit] adding state 0x7fcba8004b00
[DEBUG 14:55:41.242687] [PackageKit] role now search-name
[DEBUG 14:55:41.844934] [PackageKit] remove state 0x7fcbac0058f0
[DEBUG 14:55:41.846281] [PackageKit] adding state 0x7fcbac0059b0
[DEBUG 14:55:41.846324] [PackageKit] role now search-name
[DEBUG 14:55:42.443738] [PackageKit] remove state 0x7fcbac005ef0
[DEBUG 14:55:42.445245] [PackageKit] adding state 0x7fcbb8015810
[DEBUG 14:55:42.445708] [PackageKit] role now search-name
[DEBUG 14:55:43.273579] [PackageKit] remove state 0x7fcba8004b00
[DEBUG 14:55:43.273683] [PackageKit] adding state 0x7fcba8004bc0
[DEBUG 14:55:43.273727] [PackageKit] role now get-details
[DEBUG 14:55:44.024787] [PackageKit] remove state 0x7fcbac0059b0
[DEBUG 14:55:44.024891] [PackageKit] adding state 0x7fcbac005a70
[DEBUG 14:55:44.024933] [PackageKit] role now get-details
[DEBUG 14:55:44.801291] [PackageKit] remove state 0x7fcbb8015810
[DEBUG 14:55:44.801445] [PackageKit] adding state 0x7fcbb80158d0
[DEBUG 14:55:44.801511] [PackageKit] role now get-details
[DEBUG 14:55:45.437913] [PackageKit] remove state 0x7fcba8004bc0
[DEBUG 14:55:45.438583] [PackageKit] adding state 0x7fcba8004c80
[DEBUG 14:55:45.438615] [PackageKit] role now search-name
[DEBUG 14:55:46.157783] [PackageKit] remove state 0x7fcbac005a70
[DEBUG 14:55:46.158666] [PackageKit] adding state 0x7fcbac005b30
[DEBUG 14:55:46.158745] [PackageKit] role now search-name
[DEBUG 14:55:46.808778] [PackageKit] remove state 0x7fcbb80158d0
[DEBUG 14:55:46.809093] [PackageKit] adding state 0x7fcbb8015990
[DEBUG 14:55:46.809238] [PackageKit] role now search-name
[DEBUG 14:55:47.536076] [PackageKit] remove state 0x7fcba8004c80
[DEBUG 14:55:47.536178] [PackageKit] adding state 0x7fcba8004d40
[DEBUG 14:55:47.536223] [PackageKit] role now get-details
[DEBUG 14:55:47.581623] [PackageKit] adding state 0x7fcbb8015b10
[DEBUG 14:55:47.581688] [PackageKit] role now search-name
[DEBUG 14:55:48.159196] [PackageKit] remove state 0x7fcbac005b30
[DEBUG 14:55:48.159297] [PackageKit] adding state 0x7fcbac005bf0
[DEBUG 14:55:48.159340] [PackageKit] role now get-details
[DEBUG 14:55:48.792126] [PackageKit] remove state 0x7fcbb8015990
[DEBUG 14:55:48.792230] [PackageKit] adding state 0x7fcbb8015a50
[DEBUG 14:55:48.792273] [PackageKit] role now get-details
[DEBUG 14:55:49.354934] [PackageKit] remove state 0x7fcba8004d40
[DEBUG 14:55:49.355969] [PackageKit] adding state 0x7fcba8004d40
[DEBUG 14:55:49.356001] [PackageKit] role now search-name
[DEBUG 14:55:49.982676] [PackageKit] remove state 0x7fcbb8015b10
[DEBUG 14:55:49.982785] [PackageKit] adding state 0x7fcbb8015bd0
[DEBUG 14:55:49.982824] [PackageKit] role now get-details
[DEBUG 14:55:50.518714] [PackageKit] remove state 0x7fcbac005bf0
[DEBUG 14:55:50.520034] [PackageKit] adding state 0x7fcbac005bf0
[DEBUG 14:55:50.520110] [PackageKit] role now search-name
[DEBUG 14:55:51.060822] [PackageKit] remove state 0x7fcbb8015a50
[DEBUG 14:55:51.062230] [PackageKit] adding state 0x7fcbb8015a50
[DEBUG 14:55:51.062299] [PackageKit] role now search-name
[DEBUG 14:55:51.674690] [PackageKit] remove state 0x7fcba8004d40
[DEBUG 14:55:51.674796] [PackageKit] adding state 0x7fcba8004c80
[DEBUG 14:55:51.674834] [PackageKit] role now get-details
[DEBUG 14:55:52.230069] [PackageKit] remove state 0x7fcbb8015bd0
[DEBUG 14:55:52.230390] [PackageKit] adding state 0x7fcbb8015c90
[DEBUG 14:55:52.230452] [PackageKit] role now search-name
[DEBUG 14:55:52.869462] [PackageKit] remove state 0x7fcbac005bf0
[DEBUG 14:55:52.869550] [PackageKit] adding state 0x7fcbac005b30
[DEBUG 14:55:52.869583] [PackageKit] role now get-details
[DEBUG 14:55:53.513810] [PackageKit] remove state 0x7fcbb8015a50
[DEBUG 14:55:53.513910] [PackageKit] adding state 0x7fcbb8015990
[DEBUG 14:55:53.513950] [PackageKit] role now get-details
[DEBUG 14:55:54.040187] [PackageKit] remove state 0x7fcba8004c80
[DEBUG 14:55:54.073191] Client.vala:513: update cache called false
[DEBUG 14:55:54.073248] Client.vala:539: New refresh task
[DEBUG 14:55:54.074073] [PackageKit] adding state 0x3b80d50
[DEBUG 14:55:54.074127] [PackageKit] role now refresh-cache
[DEBUG 14:55:54.783632] [PackageKit] remove state 0x7fcbb8015c90
[DEBUG 14:55:54.783738] [PackageKit] adding state 0x7fcbb8015d50
[DEBUG 14:55:54.783784] [PackageKit] role now get-details
[DEBUG 14:55:55.319281] [PackageKit] remove state 0x7fcbac005b30
[DEBUG 14:55:55.883785] [PackageKit] remove state 0x7fcbb8015990
[DEBUG 14:56:07.142772] [PackageKit] remove state 0x3b80d50
[FATAL 14:56:07.142886] Client.vala:549: Update_cache: Refesh cache async failed - W: http://i-hate-farms.github.io/spores/dists/trusty/InRelease: Signature by key 2F3C3D45B6C9A4E9E22A1AE1F7F707E5B5329C31 uses weak digest algorithm (SHA1)W: The repository 'http://ppa.launchpad.net/mpstark/elementary-tweaks-daily/ubuntu xenial Release' does not have a Release file.W: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.W: See apt-secure(8) manpage for repository creation and user configuration details.
[DEBUG 14:56:10.001936] [PackageKit] remove state 0x7fcbb8015d50
[DEBUG 14:56:10.002170] [PackageKit] adding state 0x7fcbb8015e10
[DEBUG 14:56:10.002202] [PackageKit] role now search-name
[DEBUG 14:56:10.722452] [PackageKit] remove state 0x7fcbb8015e10
[DEBUG 14:56:10.722800] [PackageKit] adding state 0x7fcbb8015ed0
[DEBUG 14:56:10.722843] [PackageKit] role now get-details
[DEBUG 14:56:11.315494] [PackageKit] remove state 0x7fcbb8015ed0
[DEBUG 14:56:11.315769] [PackageKit] adding state 0x7fcbb4004040
[DEBUG 14:56:11.315821] [PackageKit] role now search-name
[DEBUG 14:56:11.978257] [PackageKit] remove state 0x7fcbb4004040
[DEBUG 14:56:11.978365] [PackageKit] adding state 0x7fcbb4004100
[DEBUG 14:56:11.978407] [PackageKit] role now get-details
[DEBUG 14:56:12.605184] [PackageKit] remove state 0x7fcbb4004100
</pre></code>
</details>
<details>
<summary><b>Output of <code>sudo apt upgrade</code></b></summary>
<pre><code>
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following package was automatically installed and is no longer required:
  libllvm3.8
Use 'sudo apt autoremove' to remove it.
The following NEW packages will be installed:
  libinput-bin linux-headers-4.4.0-89 linux-headers-4.4.0-89-generic linux-image-4.4.0-89-generic linux-image-extra-4.4.0-89-generic
  linux-signed-image-4.4.0-89-generic
The following packages will be upgraded:
  appcenter apt-transport-https audience binutils code com.github.davidmhewitt.clipped com.github.lainsce.notejot com.github.philip-scott.spice-up
  elementary-default-settings elementary-dpms-helper elementary-printer-test-page gdb gdbserver grub-common grub-efi-amd64 grub-efi-amd64-bin grub-efi-amd64-signed
  grub2-common heroku imagemagick imagemagick-6.q16 imagemagick-common kmod libgl1-mesa-dev libgl1-mesa-glx libglapi-mesa libgles2-mesa libgles2-mesa-dev
  libimage-magick-perl libimage-magick-q16-perl libinput-dev libinput10 libjavascriptcoregtk-4.0-18 libkmod2 libmagick++-6.q16-5v5 libmagickcore-6.q16-2
  libmagickcore-6.q16-2-extra libmagickwand-6.q16-2 libruby2.3 libswitchboard-2.0-0 libswitchboard-2.0-dev libudev-dev libudev1 libuv1 libuv1-dev libwacom-bin
  libwacom-common libwacom2 libwebkit2gtk-4.0-37 libwebkit2gtk-4.0-37-gtk2 libwhoopsie0 libxatracker2 libxfont1 linux-generic linux-headers-generic
  linux-image-generic linux-libc-dev linux-signed-generic linux-signed-image-generic mesa-common-dev mesa-vdpau-drivers neovim neovim-runtime pantheon-photos
  pantheon-photos-common pantheon-terminal plymouth-theme-elementary plymouth-theme-elementary-text python3-distupgrade python3-update-manager ruby2.3
  screenshot-tool shim-signed sudo switchboard ubiquity-slideshow-elementary ubuntu-release-upgrader-core udev unattended-upgrades update-manager-core whoopsie
  wingpanel-indicator-notifications x11proto-core-dev xfonts-utils
84 upgraded, 6 newly installed, 0 to remove and 0 not upgraded.
23 not fully installed or removed.
Need to get 0 B/177 MB of archives.
After this operation, 300 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
</pre></code>
</details>

**Comment by jwinnie** [comment]
I think I found the problem in `src/Core/UpdateManager.vala`. The `get_updates_async ()` call is wrapped in a `try` clause that throws an error if it fails, which causes this error in the debug log:
```
[FATAL 14:53:55.456107] Client.vala:549: Update_cache: Refesh cache async failed - W: http://i-hate-farms.github.io/spores/dists/trusty/InRelease: Signature by key 2F3C3D45B6C9A4E9E22A1AE1F7F707E5B5329C31 uses weak digest algorithm (SHA1)W: The repository 'http://ppa.launchpad.net/mpstark/elementary-tweaks-daily/ubuntu xenial Release' does not have a Release file.W: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.W: See apt-secure(8) manpage for repository creation and user configuration details.
```
Maybe it would be more user-friendly if AppCenter showed a `Gtk.InfoBar` that said something like this:
```
Error: Something went wrong when checking for updates. [Show Full Error Log] [Report]
```

**Comment by jtrees** [comment]
I am also encountering this issue with AppCenter 0.2.6.

**Comment by Andrewsudarikov** [comment]
Same issue with me. Restarting helps, but when new updates and apps arrive, AppCenter ignores them until the next reboot.

**Comment by jtrees** [comment]
@Andrewsudarikov That's interesting... Restarting has no effect for me. AppCenter appears to be permanently broken (at least until this bug gets fixed) on my system.

**Comment by peteruithoven** [comment]
@jtrees what happens when you run `apt update && apt upgrade`?

**Comment by jtrees** [comment]
It works without errors (see example [here](https://0bin.net/paste/i2d4uwM2sG+HWaS5#7twwed+r-SSBRvTDp3ZfTtkhSVe93ltq3yP/Cbau0To)).

**Comment by Andrewsudarikov** [comment]
@peteruithoven it works absolutely fine, gets the list of updated packages and installs them smoothly without a single error message.

**Comment by truscellino** [comment]
@jtrees I have exactly the same behaviour.
I had installed gradio via its ppa (which was the easiest method proposed for installation), but that method has been recently superseded by snap package. Hence the ppa is not longer maintained and apt-get update shows the corresponding error (see below), but still apt-get upgrade commands work OK, whereas appcenter stopped doing any update.

I removed the ppa from my sources list and appcenter is now handling updates properly. So I agree that hendling errors is not super robust in appcenter (as of October 2017), and prevents it from handling system upgrades. A bit dangerous!

Messages from apt-get update with ppa no longer maintained:
`Err:11 http://ppa.launchpad.net/haecker-felix/gradio-daily/ubuntu xenial Release                                         
  403  Forbidden
Get:12 http://packages.elementary.io/appcenter xenial InRelease [5,894 B]
Get:14 http://packages.elementary.io/appcenter xenial/main amd64 DEP-11 Metadata [22.6 kB]
Reading package lists... Done              
E: The repository 'http://ppa.launchpad.net/haecker-felix/gradio-daily/ubuntu xenial Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.`
